server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala

```diff
@@ -306,26 +307,26 @@ class Serializer @Inject() (mailboxIdFactory: MailboxId.Factory) {
 
   private implicit val setErrorDescriptionWrites: Writes[SetErrorDescription] = Json.valueWrites[SetErrorDescription]
 
-  private implicit val mailboxSetErrorWrites: Writes[MailboxSetError] = Json.writes[MailboxSetError]
```

Comment:
I think we introduced a specific MailboxSetError because Mailbox/set has some specific SetError:
```
The following extra SetError types are defined:

For destroy:

    mailboxHasChild: The Mailbox still has at least one child Mailbox. The client MUST remove these before it can delete the parent Mailbox.
    mailboxHasEmail: The Mailbox has at least one Email assigned to it, and the onDestroyRemoveEmails argument was false.
```

Reply:
Good catch ! but the generic errors it can stay extracted though I guess?

Reply:
yes probably

Reply:
I kept a `MailboxSetError with factory methods for these properties :-)`